linkml_runtime/utils/schemaview.py

```diff
@@ -104,6 +105,19 @@ def is_absolute_path(path: str) -> bool:
     drive, tail = os.path.splitdrive(norm_path)
     return bool(drive and tail)
 
+def resolve_import(sn: str, i: str) -> str:
+    if is_absolute_path(i):
+        return i
+    if urlparse(i).scheme:
+        return i
+    else:
+        path = os.path.normpath(str(Path(sn).parent / i))
+        if i.startswith(".") and not path.startswith("."):
+            # Above condition handles cases where both sn and i are relative paths, and path is not already relative
+            return f"./{path}"
+        else:
+            return path
+
```

Comment:
since we are already treating everything that isn't absolute or has a scheme as a path, i think this would be double-fixing the problem. e.g. tests pass with this

```suggestion
        return os.path.normpath(str(Path(sn).parent / i))

```

Reply:
This was here to retain compatibility with hard-coded paths in `test_imports_relative`. If removed, following error raised:

```
test_utils/test_schemaview.py::test_imports_relative failed: def test_imports_relative():
...
>       assert closure == [
            'linkml:types',
            '../neighborhood_parent',
            'neighbor',
            '../parent',
            '../L1_0_1/L2_0_1_0/grandchild',
            '../../L0_1/L1_1_0/L2_1_0_0/apple',
            '../../L0_1/L1_1_0/L2_1_0_0/index',
            '../../L0_1/L1_1_0/L2_1_0_1/banana',
            '../../L0_1/L1_1_0/L2_1_0_1/index',
            '../../L0_1/L1_1_0/index',
            '../../L0_1/cousin',
            '../L1_0_1/dupe',
            './L2_0_0_0/child',
            './L2_0_0_1/child',
            'main'
        ]
E       AssertionError: assert ['linkml:type...0/apple', ...] == ['linkml:type...0/apple', ...]
E         
E         At index 12 diff: 'L2_0_0_0/child' != './L2_0_0_0/child'
E         Use -v to get more diff
```

Reply:
ah yes. I do think it's useful to have all paths, internally in the `imports_closure` method, have a `./` or `../` in them so we can determine which need directory traversal trivially, but also acknowledge that would probably be done better with proper typing rather than string conventions. Maybe changing the comment above to something like "ensure any path that requires directory traversal has a ./ prepended" or something like that would be good, but since the above is so anticipatory of future needs (e.g. resolving remote imports, resolving content-ID imports, other contexts where we need to differentiate posix-like directories from other formats) that it borders on YAGNI so i could go either way on what happens with this. thx for explaining